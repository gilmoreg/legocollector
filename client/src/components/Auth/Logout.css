.Logout {
  padding: 0.25rem;
}

.Logout span {
  color: #2185C5;
}

.Logout button {
  display: block;
  background-color: #FF7F66;
  border: none;
  border-radius: 5px;
  margin: 0.5rem auto 0 auto;
  font-size: 12px;
}