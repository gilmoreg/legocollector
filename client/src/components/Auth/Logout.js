import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { logout } from '../../state/actions';

const StyledDiv = styled.div`
  padding: 0.25rem;
`;

const StyledSpan = styled.span`
  color: #2185C5;
`;

const StyledButton = styled.button`
  display: block;
  background-color: #FF7F66;
  border: none;
  border-radius: 5px;
  margin: 0.5rem auto 0 auto;
  font-size: 12px;
`;

export class Logout extends Component {
  constructor(props) {
    super(props);
    this.logout = this.logout.bind(this);
  }

  logout() {
    window.localStorage.removeItem('legocollectorProfile');
    this.props.dispatch(logout());
  }

  render() {
    return (
      <StyledDiv>
        <StyledSpan>Logged in as {this.props.email}</StyledSpan>
        <StyledButton onClick={this.logout}>Logout</StyledButton>
      </StyledDiv>
    );
  }
}

Logout.defaultProps = {
  email: '',
};

Logout.propTypes = {
  email: PropTypes.string,
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  email: state.email,
});

export default connect(mapStateToProps)(Logout);
