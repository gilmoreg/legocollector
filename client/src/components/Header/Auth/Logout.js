import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { logout } from '../../../state/actions';
import './Logout.css';

export class Logout extends Component {
  constructor(props) {
    super(props);
    this.logout = this.logout.bind(this);
  }

  logout() {
    window.localStorage.removeItem('legocollectorProfile');
    this.props.dispatch(logout());
  }

  render() {
    return (
      <div className="Logout">
        <span>Logged in as {this.props.email}</span>
        <button onClick={this.logout}>Logout</button>
      </div>
    );
  }
}

Logout.defaultProps = {
  email: '',
};

Logout.propTypes = {
  email: PropTypes.string,
  dispatch: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  email: state.email,
});

export default connect(mapStateToProps)(Logout);
