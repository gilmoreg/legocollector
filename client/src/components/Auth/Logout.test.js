import React from 'react';
import { shallow } from 'enzyme';
import { Logout } from './Logout';

it('renders without crashing', () => {
  const wrapper = shallow(<Logout dispatch={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

it('logs out without crashing', () => {
  window.localStorage = { removeItem: jest.fn() };
  const wrapper = shallow(<Logout dispatch={jest.fn()} />);
  wrapper.instance().logout();
  expect(window.localStorage.removeItem).toHaveBeenCalled();
  expect(wrapper.instance().props.dispatch).toHaveBeenCalled();
  window.localStorage = undefined;
});
