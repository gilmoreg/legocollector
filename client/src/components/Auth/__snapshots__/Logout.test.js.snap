// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Logout
    dispatch={[Function]}
    email=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          Logged in as 
          
        </span>,
        <button
          onClick={[Function]}
        >
          Logout
        </button>,
      ],
      "className": "Logout",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Logged in as ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Logged in as ",
          "",
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": "Logout",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Logout",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            Logged in as 
            
          </span>,
          <button
            onClick={[Function]}
          >
            Logout
          </button>,
        ],
        "className": "Logout",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Logged in as ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Logged in as ",
            "",
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Logout",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Logout",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
